src/Menu/menu/sizeModifier.ts

```diff
@@ -0,0 +1,292 @@
+      (overflow) => overflow.placement === a.overflowDict.placement
+    );
+    const bPlacementWeight = overflowDicts.findIndex(
+      (overflow) => overflow.placement === b.overflowDict.placement
+    );
+
+    return aPlacementWeight - bPlacementWeight;
+  });
+
+  return overflowsWithOverflowMap[0];
+}
+
+/**
+ * Padding can be applied to `preventOverflow`, `flip`, or this modifier. Go
+ * through all the modifier options to find it
+ */
+function getPaddingFromState(
+  state: ModifierArguments<Options>["state"]
+): Padding {
+  return state.orderedModifiers.reduce<Padding>((accumulator, modifier) => {
+    if (typeof modifier.options?.padding != "undefined") {
+      accumulator = modifier.options.padding;
+    }
+
+    return accumulator;
+  }, 0);
+}
+
+function getPlacementAndMaxSize({
+  state,
+  options,
+}: ModifierArguments<Options>): {
+  placement: Placement;
+  maxSize: { height: number };
+} {
+  const {
+    fallbackPlacements: specifiedFallbackPlacements,
+    padding = getPaddingFromState(state),
+    boundary = "clippingParents",
+    rootBoundary = "viewport",
+    altBoundary,
+    flipVariations = true,
+    allowedAutoPlacements,
+  } = options;
+
+  /**
+   * Preferred placement
+   *
+   * Taken from options
+   */
+  const preferredPlacement = state.options.placement;
+
+  /**
+   * Base placement from `preferredPlacement`
+   */
+  const basePlacement = getBasePlacement(preferredPlacement);
+
+  /**
+   * Represents if the `preferredPlacement` is a `BasePlacement` (meaning it has
+   * no `-begin` or `-end`)
+   */
+  const isBasePlacement = basePlacement === preferredPlacement;
+
+  /**
+   * List of fallback placements
+   *
+   * Either passed in with `specifiedFallbackPlacements`, or calculated
+   * depending on
+   * [`flipVariations`](https://popper.js.org/docs/v2/modifiers/flip/#flipvariations)
+   */
+  const fallbackPlacements =
+    specifiedFallbackPlacements ||
+    (isBasePlacement || !flipVariations
+      ? [getOppositePlacement(preferredPlacement)]
+      : getExpandedFallbackPlacements(preferredPlacement));
+
+  /**
+   * Aggregate of all placements, including preferred and fallbacks.
+   */
+  const placements = [preferredPlacement, ...fallbackPlacements].reduce<
+    Placement[]
+  >((acc, placement) => {
+    return acc.concat(
+      getBasePlacement(placement) === "auto"
+        ? computeAutoPlacement(state, {
+            placement,
+            boundary,
+            rootBoundary,
+            padding,
+            flipVariations,
+            allowedAutoPlacements,
+          })
+        : placement
+    );
+  }, []);
+
+  const referenceRect = state.rects.reference;
+  const popperRect = state.rects.popper;
+
+  /**
+   * Array calculated from `placements` with the calculated values of each
+   * `placement`. This will be used to determine if we're capable of flipping
+   * the element to display it ir of we have to set the `max-height` too.
```

Comment:
Is this comment still accurate? This doesnt seem like what we really end up using this for?

Reply:
Updated the comment; it was still accurate but the logic was convoluted.